danny-avila/LibreChat
# 👮feat: moderation text

## Summary

Integrate text moderation middleware with OpenAI API for content filtering
It works only with OpenAI or GPTPlugins

### text detected

![image](https://github.com/danny-avila/LibreChat/assets/81851188/c171909a-96e9-4df0-8efb-da39a6a59c2e)

### error with the moderation check (POST request failed, API-key error, bad URL etc...)

![image](https://github.com/danny-avila/LibreChat/assets/81851188/03e89db2-509f-42b5-b48f-387018ca4f35)


## Change Type


## Testing

1. Set up environment variables:
```
OPENAI_MODERATION should be set to 'true' to activate moderation
OPENAI_MODERATION_REVERSE_PROXY (optional) can be set to a custom moderation API endpoint.
OPENAI_MODERATION_API_KEY should contain a valid API key for OpenAI's moderation service.
```
2. Create test cases with various text inputs, including both clean and inappropriate content
3. Send HTTP requests with test cases to the server

## Checklist
